hid-xpadneo/src/hid-xpadneo.c

```diff
@@ -112,23 +112,33 @@ struct usage_map {
 #define USAGE_IGN(u) USAGE_MAP(u, MAP_IGNORE, 0, 0)
 
 static const struct usage_map xpadneo_usage_maps[] = {
+  // 90009 and 9000a, sent when trigger fully released
 	/* fixup buttons to Linux codes */
 	USAGE_MAP(0x90001, MAP_STATIC, EV_KEY, BTN_A),	/* A */
 	USAGE_MAP(0x90002, MAP_STATIC, EV_KEY, BTN_B),	/* B */
-	USAGE_MAP(0x90003, MAP_STATIC, EV_KEY, BTN_X),	/* X */
```

Comment:
Looking at pictures of the device, it has two additional buttons at the center (house and circle), and also buttons C and Z. These are not part of the Xbox spec and must be moved outside the range of the first 10 buttons: The Xbox spec supports exactly 10 well defined buttons in exactly that order: A,B,X,Y,LB,RB,Select,Start,LS,RS - no more, and none inserted. There's an additional Guide button at position 11 - but it's actually not meant to be dealt with by games, thus it's not part of the 10 well defined buttons. The Windows XINPUT spec (and thus what Proton uses) supports exactly 16 buttons maximum, no more. So every excess button in the keymap must be masked.

So either your device MUST not report as an Xbox controller, or exactly match those expectations.

You can patch the HID descriptors and HID reports to move the additional buttons to the `BTN_TRIGGER_HAPPY` range. We already moved some buttons there, especially the paddles have been moved there.

**Other than that, I can only repeat:** DO NOT misuse xpadneo to support this device if you don't take the above comments serious. DO NOT fork this project to support the controller, if you do not remove the VID/PID patching and do not remove support for MS devices. Really, I DO NOT want this device show up in random SDL controllerdb mappings destroying all the effort we've put into unifying all Xbox devices into a single virtual model currently properly supported by SDL and Steam.

Everything you do is reverting all these efforts and making the device show up like it would do with the hid-generic driver probably, but then probably (mis-)use our VID/PID patching technique to make the device visible to SDL and games as an Xbox controller - which it isn't in HID mode (aka Bluetooth). Instead, stick to hid-generic and add a SDL controllerdb mapping to make it visible in Steam and games. Ask the SDL project to properly support your model with hid-generic. It may be the easier way.

Please, do NOT submit a controllerdb patch to SDL while using this Frankenstein patched xpadneo driver.

Do not understand me wrong: If done properly, we could support your device in xpadneo. But it is important to know the consequences of doing it the wrong way. A lot of effort went into getting to the point where we are now.

See also #286. This is also about how we should probably avoid HID patching in the future and thus stop VID/PID patching, too. That change would then render your controller useless with xpadneo because your initial idea of using xpadneo is probably because games do not see `D7D7:0041` as an Xbox controller. OTOH, there's an idea of xpadneo converting other supported controllers to Xbox controllers (e.g. Switch mode of the Gulikit controllers). In this case, your controller would fit the bill of xpadneo. But it MUST be done properly: No changes to the 10 first buttons in the map, no changes to the axes.